app/src/main/java/com/eveningoutpost/dexdrip/Models/JoH.java

```diff
@@ -1063,6 +1063,19 @@ public static synchronized MediaPlayer playSoundUri(String soundUri) {
             JoH.getWakeLock("joh-playsound", 10000);
             final MediaPlayer player = MediaPlayer.create(xdrip.getAppContext(), Uri.parse(soundUri));
             player.setLooping(false);
+            player.setOnCompletionListener(mp -> {
+                UserError.Log.i(TAG, "playSoundUri: onCompletion called (finished playing) ");
+                try {
```

Comment:
```java
                player.release();
```

Should be sufficient. According to the [docs](https://developer.android.com/reference/android/media/MediaPlayer#release()), release() does not throw any exception.

Reply:
as this code is copied from AlertPlayer, i have to change it there too....